scipy/special/_logsumexp.py

```diff
@@ -114,8 +118,12 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
     with np.errstate(divide='ignore'):
         s = np.sum(tmp, axis=axis, keepdims=keepdims)
         if return_sign:
-            sgn = np.sign(s)
-            s *= sgn  # /= makes more sense but we need zero -> zero
+            # For complex, use the numpy>=2.0 convention for sign.
+            if np.issubdtype(s, np.complexfloating):
+                sgn = np.where(s == 0, 0, s / abs(s))
+            else:
+                sgn = np.sign(s)
+            s = abs(s).astype(s.dtype)
```

Comment:
Since we're slightly changing API anyway, would it make sense to not do `.astype(s.dtype)`? With the sign taken out, the answer should arguably be float even for complex.

Reply:
Sure, that's a possiblity. Let's see what the maintainers think.

Reply:
I think this is a good suggestion. Done.